src/WorkflowStep.spec.ts

```diff
@@ -0,0 +1,395 @@
+    it('configure should call views.open', async () => {
+      const fakeEditArgs = (createFakeStepEditAction() as unknown) as AllWorkflowStepMiddlewareArgs;
+
+      const fakeClient = { views: { open: sinon.spy() } };
+      fakeEditArgs.client = (fakeClient as unknown) as WebClient;
+
+      const { prepareStepArgs } = await importWorkflowStep();
+      // casting to returned type because prepareStepArgs isn't built to do so
+      const editStepArgs = prepareStepArgs(fakeEditArgs) as AllWorkflowStepMiddlewareArgs<
+        WorkflowStepEditMiddlewareArgs
+      >;
+
+      await editStepArgs.configure({ blocks: [] });
+
+      assert(fakeClient.views.open.called);
+    });
+
+    it('update should call workflows.updateStep', async () => {
+      const fakeSaveArgs = (createFakeStepSaveEvent() as unknown) as AllWorkflowStepMiddlewareArgs;
+
+      const fakeClient = { workflows: { updateStep: sinon.spy() } };
+      fakeSaveArgs.client = (fakeClient as unknown) as WebClient;
+
+      const { prepareStepArgs } = await importWorkflowStep();
+      // casting to returned type because prepareStepArgs isn't built to do so
+      const saveStepArgs = prepareStepArgs(fakeSaveArgs) as AllWorkflowStepMiddlewareArgs<
+        WorkflowStepSaveMiddlewareArgs
+      >;
+
+      await saveStepArgs.update();
+
+      assert(fakeClient.workflows.updateStep.called);
+    });
+
+    it('complete should call workflows.stepCompleted', async () => {
+      const fakeExecuteArgs = (createFakeStepExecuteEvent() as unknown) as SlackWorkflowStepMiddlewareArgs &
+        AllMiddlewareArgs;
+
+      const fakeClient = { workflows: { stepCompleted: sinon.spy() } };
+      fakeExecuteArgs.client = (fakeClient as unknown) as WebClient;
+
+      const { prepareStepArgs } = await importWorkflowStep();
+      // casting to returned type because prepareStepArgs isn't built to do so
+      const executeStepArgs = prepareStepArgs(fakeExecuteArgs) as AllWorkflowStepMiddlewareArgs<
+        WorkflowStepExecuteMiddlewareArgs
+      >;
+
+      await executeStepArgs.complete();
+
+      assert(fakeClient.workflows.stepCompleted.called);
+    });
+
+    it('fail should call workflows.stepFailed', async () => {
+      const fakeExecuteArgs = (createFakeStepExecuteEvent() as unknown) as SlackWorkflowStepMiddlewareArgs &
+        AllMiddlewareArgs;
+
+      const fakeClient = { workflows: { stepFailed: sinon.spy() } };
+      fakeExecuteArgs.client = (fakeClient as unknown) as WebClient;
+
+      const { prepareStepArgs } = await importWorkflowStep();
+      // casting to returned type because prepareStepArgs isn't built to do so
+      const executeStepArgs = prepareStepArgs(fakeExecuteArgs) as AllWorkflowStepMiddlewareArgs<
+        WorkflowStepExecuteMiddlewareArgs
+      >;
+
+      await executeStepArgs.fail({ error: { message: 'Failed' } });
+
+      assert(fakeClient.workflows.stepFailed.called);
+    });
+  });
+
+  describe('processStepMiddleware', () => {
```

Comment:
it seems like this could be another place we're testing the implementation details rather than the APIs contract, since `processStepMiddleware()` doesn't seem intended for use on its own. can we fold these assertions into another existing test case, or create new test cases in a different group that still exercise this functionality?